### ISSUE TITLE
TLSH fuzzy hash

### ISSUE THREAD

**Issue by eljeffeg** [CybercentreCanada/assemblyline#54]
TLSH is a fuzzy matching program and library. Given a file (min 50 bytes), TLSH generates a hash value which can be used for similarity comparisons. Similar files will have similar hash values which allows for the detection of similar objects by comparing their hash values TLSH has been adopted by a range of bodies and malware repositories including:

- VirusTotal (https://developers.virustotal.com/v3.0/reference#files-tlsh)
- Malware Bazaar (https://bazaar.abuse.ch/)
- MISP (https://www.misp-project.org/)
- STIX: TLSH is a part of the STIX 2.1 standard (https://docs.oasis-open.org/cti/stix/v2.1/stix-v2.1.html).

More information about TLSH, along with diff compare examples and reading in via buffer can be found at: https://tlsh.org/

Below is what we currently have for including TLSH as a Service, but we think TLSH would be a nice addition to the AssemblyLine Core to include with other hashes like SSDEEP.  Installed via [py-tlsh](https://pypi.org/project/py-tlsh/)
```py
import shutil
import tempfile
from pathlib import Path

import tlsh
from assemblyline_v4_service.common.base import ServiceBase
from assemblyline_v4_service.common.result import Result, ResultJSONSection


class TlshService(ServiceBase):

    def __init__(self, config=None):
        super(TlshService, self).__init__(config)

    def start(self):
        self.log.info(f"start() from {self.service_attributes.name} service called")

    def execute(self, request):
        self.log.info(f"execute() from {self.service_attributes.name} service called for '{request.file_name}'")

        try:
            with open(request.file_path, "rb") as sample_file_object:
                working_dir = Path(tempfile.mkdtemp())

                try:
                    sample_bytes = sample_file_object.read()
                    tlsh_digest = tlsh.hash(sample_bytes).lower()

                    result = Result()

                    json_section = ResultJSONSection("TLSH Response")
                    json_section.set_json({'tlsh': tlsh_digest})
                    result.add_section(json_section)

                    request.result = result
                finally:
                    shutil.rmtree(working_dir, ignore_errors=True)

            result = Result()

        except Exception as ex:
            self.log.error(str(ex))
            raise
```

**Comment by cccs-sgaron** [comment]
We already have an internal ticket for this, we'll add it soon directly in the file index.

**Comment by cccs-sgaron** [comment]
Fixed by: https://github.com/CybercentreCanada/assemblyline-base/pull/1205

**Comment by cccs-rs** [comment]
Fixed in [Release 4.4.0.21](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.4.0.stable21)
